### ISSUE TITLE
[SERVICE] backup and restore options

### ISSUE THREAD

**Issue by RobertHosking** [project-rainstorm/project_rainstorm#11]
What we're building is great until the HDD fails and you realize that your entire digital life has been erased.

With great power, comes great responsibility. But we're all about making shit easier.

Let's use [restic](https://restic.net/) for backups and leverage this [tardigrade connector](https://documentation.tardigrade.io/how-tos/backup-with-restic).

This combo provides a highly efficient backup process that stores data in the public/decentralized cloud.

Storing data on Tardigrade does incur cost. Its still quite cheap compared to AWS so we can afford to bank roll it ourselves for a while. Long-term, this is premium feature.

**Comment by nullcount** [comment]
Using tardigrade requires an access token which grants permissions to a specified key in the bucket. I'm able to generate tokens with `uplink access sj://dropcloud/<user-backup-key>` on my instance of lib-uplink configured with the Tardigrade API key.

We need a central micro-service which accepts requests for tokens and returns a token. The endpoint `POST /backup-access/<user-key>` should validate the key (i.e. is the key belonging to a premium user) and return a valid token for the `sj://dropcloud/<user-key>` object.

**Comment by newswangerd** [comment]
@count-null how many buckets can you create on tardigrade? B2 is limited to 100 per account

**Comment by nullcount** [comment]
@newswangerd I encountered a potential issue with restic. Backups are encrypted with a password. Since all the calls to restic are made in scripts, we can't have the user entering their password in the terminal like restic wants. Thankfully, restic has a `--password-file` flag that accepts the password stored in a file. 

This works great except that when we change the password for restic, it requires you to type in the new password. There is no flag for that. Maybe it's for security, maybe its just an oversight. I made an issue on the restic/restic repo. It's written in Go.

We could get hacky with it and create a new restic repo when the user changes their password. But it's worth exploring if there are other backup options that integrate with rclone.

We could always just use regular rclone to tardigrade. Restic is nice because it dedupes and encrypts everything.
